### PR TITLE
ft(buses):view-list-of-buses-rIMSneoO

### PR DESCRIPTION
**Pull request by Ntwali-Josue**
**What about this PR?**
This PR has a route that enables the user to View list of buses in a certain route

**Description of Task to be completed**
It should have the following endpoint:

- `The user should be able to provide the basic selection criteria i.e. origin & destination location`
- `The user should be able to see an ordered list of buses`

**How should be manually tested?**

You may use [swagger documentation](https://phanton-back-view-list--tzikc6.herokuapp.com/api-docs/#/Buses/get_api_v1_bus_routes_origin__origin__destination__destination_) under the endpoint provided before Or

You may also use postman with this link of [view list of buses in a route](https://phanton-back-view-list--tzikc6.herokuapp.com/api/v1/bus/routes?origin=down-town&destination=Kabeza) 

provide your origin and destination and there values in params

**Relevant pivots tracker**
[#rIMSneoO](https://trello.com/c/rIMSneoO/15-view-a-list-of-buses)

### CODE REVIEW

**Review by Mugema123 — APPROVED** [review]

**Review by TuyizeeAnastase — APPROVED** [review]

**Review by Ally4 — APPROVED** [review]

File: .eslintrc.json [hunk 1]
```diff
@@ -13,6 +13,9 @@
     },
     "rules": {
         "no-underscore-dangle":"off",
-        "import/no-extraneous-dependencies":["error", {"devDependencies": false, "optionalDependencies": false, "peerDependencies": false}]
+        "import/no-extraneous-dependencies":["error", {"devDependencies": false, "optionalDependencies": false, "peerDependencies": false}],
+        "consistent-return": "off",
+        "radix": "off",
+        "no-console": "off"
```
Comment (ceelogre) [inline .eslintrc.json:19]: Why did you add new rules to eslint?

File: src/controllers/bus.controller.js [hunk 2]
```diff
@@ -2,8 +2,9 @@ import dotenv from 'dotenv';
 import mail from '@sendgrid/mail';
 import { paginate } from 'paginate-info';
 import model from '../database/models';
-import { messageBus } from '../utils/busMails';
```
Comment (ceelogre) [inline src/controllers/bus.controller.js:5]: This file is too long. Consider refactoring.

File: src/database/models/bus.js [hunk 3]
```diff
@@ -15,20 +15,21 @@ module.exports = (sequelize, DataTypes) => {
...
+
   Bus.init({
     busPlate: DataTypes.STRING,
     busStatus: DataTypes.STRING,
     busLocation: DataTypes.STRING,
     busCompany: DataTypes.STRING,
     busSeats: DataTypes.STRING,
-    userId: DataTypes.STRING
+    userId: DataTypes.STRING,
+    routeId: DataTypes.INTEGER,
```
Comment (ceelogre) [inline src/database/models/bus.js:29]: Does a bus has to be always associated with a route?

**Review by ceelogre — CHANGES_REQUESTED** [review]
There's no translation for other languages. Take a look at the comments too.

File: .eslintrc.json (hunk 1 above)
Comment (Ntwali-Josue) [reply .eslintrc.json:19]: Eslint was giving me more unnecessary errors, and I came up with an idea to do add them.

File: src/database/models/bus.js (hunk 3 above)
Comment (Ntwali-Josue) [reply src/database/models/bus.js:29]: yeah, I think so cause I think no buses can exist without having a route which it will be using in its activities.

**Review by ceelogre — COMMENTED** [review]
Are you sure there's translation?
Capture d’écran 2021-01-14 à 11 55 18

**Review by ceelogre — APPROVED** [review]
